### PR TITLE
chore: replace deprecated codecov uploader with GitHub action

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

fixes https://github.com/Azure/secrets-store-csi-driver-provider-azure/issues/647

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [x] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

File: .github/workflows/cover.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  codecov:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (nilekhc) [inline .github/workflows/cover.yaml:14]: Let's pin all the action to a specific SHA

**Comment by nilekhc** [comment]
Do we have a sample run for this action?

**Comment by nilekhc** [comment]
> Do we have a sample run for this action?

Got the run here:
https://github.com/Azure/secrets-store-csi-driver-provider-azure/runs/3669565302?check_suite_focus=true

File: .github/workflows/cover.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    branches:
+      - master
+
+jobs:
+  codecov:
+    runs-on: ubuntu-latest
+    steps:
+      # pinning to the sha 5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f from https://github.com/actions/checkout/releases/tag/v2.3.4
+      - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+      # pinning to the sha 331ce1d993939866bb63c32c6cbbfd48fa76fc57 from https://github.com/actions/setup-go/releases/tag/v2.1.4
```
Comment (sozercan) [inline .github/workflows/cover.yaml:16]: setup-go and codecov is not pinned but comments says so. do we want pinning here?

Comment (aramase) [reply .github/workflows/cover.yaml:16]: pushed the commit with the pinning

**Review by nilekhc — APPROVED** [review]
lgtm

**Pull request merged**